fre/yamltools/combine_yamls.py

```diff
@@ -0,0 +1,265 @@
+        combined   :  final combined file name
+        experiment :  experiment name
+        platform   :  platform used
+        target     :  targets used
+    """
+    # copy model yaml info into combined yaml
+    with open(combined,'w+',encoding='UTF-8') as f1:
+        f1.write(f'name: &name "{experiment}"\n')
+        f1.write(f'platform: &platform "{platform}"\n')
+        f1.write(f'target: &target "{target}"\n\n')
+        with open(modelyaml,'r',encoding='UTF-8') as f2:
+            f1.write("### MODEL YAML SETTINGS ###\n")
+            shutil.copyfileobj(f2,f1)
+    print(f"   model yaml: {modelyaml}")
+
+def experiment_check(mainyaml_dir,comb,experiment):
+    """
+    Check that the experiment given is an experiment listed in the model yaml.
+    Extract experiment specific information and file paths.
+    Arguments:
+        mainyaml_dir  :  model yaml file
+        comb            :  combined yaml file name
+        experiment      :  experiment name
+    """
+    comb_model=yaml_load(comb)
+
+    # Check if exp name given is actually valid experiment listed in combined yaml
+    exp_list = []
+    for i in comb_model.get("experiments"):
+        exp_list.append(i.get("name"))
+
+    if experiment not in exp_list:
+        raise NameError(f"{experiment} is not in the list of experiments")
+
+    # set platform yaml filepath
+    if comb_model["shared"]["compile"]["platformYaml"] is not None:
+        py=comb_model["shared"]["compile"]["platformYaml"]
+        py_path=Path(os.path.join(mainyaml_dir,py))
+    else:
+        py=False
+
+    # Extract compile yaml path for exp. provided
+    # if experiment matches name in list of experiments in yaml, extract file path
+    for i in comb_model.get("experiments"):
+        if experiment == i.get("name"):
+            compileyaml=i.get("compile")
+            expyaml=i.get("pp")
+            analysisyaml=i.get("analysis")
+
+            if compileyaml is not None:
+                cy_path=Path(os.path.join(mainyaml_dir,compileyaml))
+            else:
+                cy_path=False
+            if expyaml is not None:
+                ey_path=[]
+                for e in expyaml:
+                    ey=Path(os.path.join(mainyaml_dir,e))
+                    ey_path.append(ey)
+            else:
+                ey_path=False
+            if analysisyaml is not None:
+                ay_path=[]
+                for a in analysisyaml:
+                    ay=Path(os.path.join(mainyaml_dir,a))
+                    ay_path.append(ay)
+            else:
+                ay_path=False
+
+            return (py_path,cy_path,ey_path,ay_path)
+
+def combine_compile(comb_m,compileyaml):
+    """
+    Combine compile yaml with the defined combined.yaml
+    Arguments:
+        comb_m       :  combined model yaml file
+        compileyaml  :  compile yaml file 
+    """
+    combined = comb_m
+
+    # copy compile yaml info into combined yaml
+    if compileyaml is not False:
```

Comment:
worth noting in this function if `compileyaml` is a string representing a path to a `yaml`, or if it is a loaded `yaml` object itself... or perhaps, it's a boolean?

Since later lines imply `compileyaml` is a string, `if compileyaml is not None:` will be preferable.

Reply:
Initially, I had these as `is not None` but I noticed issues with the analysis script merge section. The analysis scripts are not fully developed yet but we think they will act the same as  the experiment pp yamls (listed under each experiment if created). Soo, when the script looks for `analysisyaml=i.get("analysis")` that isn't there yet, it would say 
```
UnboundLocalError: local variable 'ay_path' referenced before assignment
```
Paired with a pylint message that I can't even remeber anymore, I think at one point I thought I had to set the if statement to something like `if analysisyaml not defined, it's false`. Big oof on my part, I can just set it to `None` instead of `False` and keep what I originally had with `is not None`. 

I'll fix this, thanks!